runtime/vm/classsupport.c

```diff
@@ -227,6 +227,13 @@ internalCreateArrayClassWithOptions(J9VMThread *vmThread, J9ROMArrayClass *romCl
 			}
 		}
 	}
+
+	/* Create elementClass's arrayClass as a prerequisite to nullRestrictedArrayClass */
+	if (J9_ARE_ALL_BITS_SET(options, J9_FINDCLASS_FLAG_CLASS_OPTION_NULL_RESTRICTED_ARRAY)
+		&& (NULL == elementClass->arrayClass)
+	) {
+		internalCreateArrayClass(vmThread, romClass, elementClass);
```

Comment:
Currently `internalCreateArrayClassWithOptions()` calls into itself recursively. Maybe you can rename this function to `internalCreateArrayClassHelper()` and let `internalCreateArrayClass()` and `internalCreateArrayClassWithOptions()` call this helper function, so that we can avoid the recursion. (Call `internalCreateArrayClassHelpers()` twice if  `J9_FINDCLASS_FLAG_CLASS_OPTION_NULL_RESTRICTED_ARRAY` is set, one to create normal array, one to create null-restricted array)